crates/librqbit/src/session.rs

```diff
@@ -885,6 +885,9 @@ impl Session {
                     let info_hash = magnet
                         .as_id20()
                         .context("magnet link didn't contain a BTv1 infohash")?;
+                    if let Some(so) = magnet.get_select_only() {
+                        opts.only_files = Some(so);
```

Comment:
Hmmm. Soo, my problem here is it appears that the frontend always specifies "only_files" (set to everything), so with the current implementation in the frontend, it will always override the "so" in the magnet link. Any suggestions? 

for example, the API call frontend is making on a large torrent, with so set is as here:
```
/torrents?&overwrite=true&only_files=0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100&initial_peers=127.0.0.1:1234&output_folder=/tmp/scratch/files&is_url=true
```

I had a diff such as below, but it doesn't work due to the frontend overriding.

```
diff --git a/crates/librqbit/src/session.rs b/crates/librqbit/src/session.rs
index 95341d3..14729e3 100644
--- a/crates/librqbit/src/session.rs
+++ b/crates/librqbit/src/session.rs
@@ -886,7 +886,10 @@ impl Session {
                         .as_id20()
                         .context("magnet link didn't contain a BTv1 infohash")?;
                     if let Some(so) = magnet.get_select_only() {
-                        opts.only_files = Some(so);
+                        // Only overwrite opts.only_files if user didn't specify
+                        if opts.only_files.is_none() {
+                            opts.only_files = Some(so);
+                        }
                     }
 
                     let peer_rx = self.make_peer_rx(
```

Reply:
> Any suggestions?

I think fixing the backend first (as what this PR does) makes complete sense regardless.

UI frontend can be tweaked later. But with this PR alone CLI and API (both Rust and HTTP) would support "so", only frontend won't.

Feel free to update the frontend as part of this PR too though. But if you don't know how, I can take a look after this is merged

Reply:
The patch seems correct

Reply:
https://github.com/ikatson/rqbit/pull/248/commits/c555e07c08f3ccb9cd77f85e8f2e59d02ed8d49d